lincheck/src/main/java/org/jetbrains/kotlinx/lincheck/verifier/CustomScenarioDSL.kt

```diff
@@ -0,0 +1,133 @@
+/*-
+ * #%L
+ * Lincheck
+ * %%
+ * Copyright (C) 2019 JetBrains s.r.o.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.jetbrains.kotlinx.lincheck.verifier
+
+import org.jetbrains.kotlinx.lincheck.*
+import org.jetbrains.kotlinx.lincheck.execution.*
+import java.lang.IllegalStateException
+import kotlin.reflect.KFunction
+import kotlin.reflect.jvm.javaMethod
+
+/**
+ * Kotlin DSL for defining custom scenarios.
+ *
+ *
+ * Example:
+ * ```
+ * scenario {
```

Comment:
In our semantics `Operation` = `Actor` + `Result`, and custom scenarios are defined by actors only. Please, get rid of this `operation` constructor and use `actor` constructor to create actors for a scenario.
So, the dsl should look like this:
```
scenario {
     initial {
      actor(CustomTest::offer, 1)
    }
    ....
 }
```

Reply:
And fix the custom scenario dsl example in README

Reply:
I thought that `actor` is an internal name in LinCheck while `operation` is public and understandable because of `@Operation` annotation.
There are 77 mentions of `operation` in readme, but only 4 of `actor` (such as `actorsPerThread` parameter, note that its definition has word `operations`)

Reply:
From the API perspective `operation` is just a method without concrete parameters (`@Operation` marks just a method to test) whereas `actor = method + parameters`.
So, it's better to use `actor` constructor for this dsl.

Reply:
Ok